### DISCUSSION TITLE
Add flip/mirror option to GridMap

### DISCUSSION THREAD

**Discussion by matt-testing** [godotengine/godot-proposals#5878] [Engine Core]
Flip/mirror for Gridmap tiles was proposed some time ago, the issue was closed and forgotten due to the separation of "proposals" into a separate github section. However its 1. a really good idea, and 2. had quite a bit of thought already put into the implementation, apparently including actual implementation in Godot forks. Please see the closed discussion here:

https://github.com/godotengine/godot/issues/31340

Why is it a good idea? To put it simply, it reduces the number of unique tile models needed in the meshlibrary to create seamless tilesets. I can go into great detail about this if necessary, but i think its self explanatory. Please check out the linked (closed) issue, I'd love to get some feedback on whether theres interest in pursuing this - seems like a high value/low effort win-win to me.

**Comment by theraot** [comment]
For Godot 4 things have already changed somewhat in that `GridMap` now have a `get_orthogonal_index_from_basis` method that returns the `int` to use in `set_cell_item`. But it does not even consider the possibility of a mirrored `Basis`, and there is no clear way to have it return another value. Thus, I believe it is better to extend the range of values to include their mirrored counterparts.

As the linked issue mentions, only one extra bit should be needed. In this case, that means doubling the possible values from 24 to 48.

---

If it were for Godot 3, we would have more flexibility, but I believe it is better to do it the same way it is done for Godot 4. So no new parameters or fields are added. Just the list of possible values. And that also means no broken backwards compatibility.

---

And we would also have add the options to mirror the cursor to the GridMap menu. And perhaps a method to mirror a `Basis` which would make things easier to implement. And those being additions don't break compatibility either.

---

**Addendum**: There is still the normals issue.

**Comment by matt-testing** [comment]
thanks for the info, I went ahead and opened an actual proposal for this: https://github.com/godotengine/godot-proposals/issues/5883#issue-1470899819
